src/misc/ansi_code_markup.cpp

```diff
@@ -25,97 +25,213 @@
 #include "support.h"
 #include "ansi_code_markup.h"
 
-class ColorParser {
+/*!
+ * \brief Represents a markup tag.
+ *
+ * The constructor takes the string of a tag name and its optional value, and
+ * attempts to construct a valid tag that can be easily queried for its
+ * properties.
+ *
+ * The tag will be valid if, and only if, valid() returns true. Otherwise the
+ * tag could not be recognized.
+ */
+class Tag {
 public:
+	enum class TagGroup {
+		InvalidGroup,
+		Colors,
+		Erasers,
+		Styles,
+		Misc,
+	};
+
+	enum class TagName {
+		InvalidName,
+		Color,
+		BGColor,
+		EraseL,
+		EraseS,
+		It,
+		Bold,
+		Ul,
+		Strike,
+		Blink,
+		Dim,
+		Hidden,
+		Inverse,
+		Reset,
+	};
+
+	enum class ColorName {
+		InvalidColor,
```

Comment:
The ColorName -> InvalidColor doesn't fit the other patterns:
 - TagGroup -> InvalidGroup
 - TagName -> InvalidName
 - EraseExtents -> InvalidExtents

This might indicate either TagName or ColorName should be changed. I would say that `Color::Blue` reads more intuitively than `ColorName::Blue`.  

As for TagName - I think `Name` might be losing some of the meaning. This collection represents the _type of formatting change_ or _action_ I want to apply to the text.